src/popup/utils/utils.ts

```diff
@@ -704,3 +706,12 @@ export function checkImageAvailability(url: string): Promise<boolean> {
     .then((response) => !!response.ok)
     .catch(() => false);
 }
+
+export function deriveProtocolFromAddress(address: string): Protocol {
+  switch (address.charAt(0).toLowerCase()) {
+    case 'a': return PROTOCOLS_SUPPORTED[0];
+    case 'b': return PROTOCOLS_SUPPORTED[1];
+    case 'd': return PROTOCOLS_SUPPORTED[2];
+    default: throw new Error('Unsupported protocol');
+  }
```

Comment:
In my opinion checking only the first letter of the `address` is not a good way to detect the protocol. For the AE account addresses we have the `checkAddress` utility function.

Reply:
This code is no longer needed hence removed.